asapdiscovery-ml/asapdiscovery/ml/weights/weights.py

```diff
@@ -0,0 +1,140 @@
+            )
+        filename_set.add(weights)
+    return weights_files
+
+
+def fetch_weights(weights: str, path: str = None, force_fetch: bool = False) -> Path:
+    """Fetch weights from remote and save to local path.
+
+    Parameters
+    ----------
+    weights : str
+        Remote url or local path to weights.
+    path : str, optional
+        Local path to save weights if a remote url is provided. or to check if weights exist locally, by default None
+    force_fetch : bool, optional
+        Force fetch weights from remote, by default False
+
+    Returns
+    -------
+    Path to weights file.
+    """
+
+    is_url = validators.url(weights)
+
+    if is_url:
+        logging.info("Weights are a remote url")
+        logging.info(f"Fetching weights from {weights}")
+        if force_fetch:
+            logging.info("Force fetching weights from remote")
+            if not Path(path).exists():
+                logging.info(f"Local path {path} does not exist, creating")
+                Path(path).mkdir(parents=True, exist_ok=True)
+            # fetch from remote
+            weights_file = download_file(weights, path)
+            return weights_file
+        else:
+            # check if weights exist locally
+            local_filename = weights.split("/")[-1]
+            if Path(path + local_filename).exists():
+                # exists locally, skip fetch from remote
+                logging.info(
+                    f"weights exist locally at {path + local_filename}, skipping fetch from remote"
+                )
+                return Path(path + local_filename)
+            else:
+                # fetch from remote
+                logging.info("weights do not exist locally, fetching from remote")
+                if not Path(path).exists():
+                    logging.info(f"Local path {path} does not exist, creating")
+                    Path(path).mkdir(parents=True, exist_ok=True)
+                weights_file = download_file(weights, path)
+                return weights_file
+    else:
+        logging.info(f"Fetching weights from {path + weights}")
+        if not Path(path + weights).exists():
+            raise FileNotFoundError(
+                f"Local weights file {path + weights} does not exist, provide a valid remote url or local path to weights"
+            )
+
+        else:
+            logging.info(
+                f"weights exist locally at {path + weights}, skipping fetch from remote"
+            )
+            return Path(path + weights)
+
+
+# https://stackoverflow.com/questions/16694907/
+def download_file(url: str, path: str) -> None:
+    """Download file from url and save to path.
+
+    Parameters
+    ----------
+    url : str
+        Remote url to download file from.
+    path : str
+        Local path to save file to.
+    """
+    local_filename = url.split("/")[-1]
+    path = Path(path + local_filename)
```

Comment:
maybe add a check here for if `path` is a file vs directory, eg
```suggestion
    if os.path.isdir(path):
        local_filename = url.split("/")[-1]
        path = Path(path + local_filename)
    else:
        path = Path(path)
```